cg/meta/transfer/flowcell.py

```diff
@@ -20,111 +46,154 @@ def __init__(self, db: Store, stats_api: StatsAPI, hk_api: HousekeeperAPI):
+    def transfer(self, flow_cell_dir: Path, flow_cell_id: str, store: bool = True) -> Flowcell:
         """Populate the database with the information."""
-        if store and self.hk.tag("fastq") is None:
-            self.hk.add_commit(self.hk.new_tag("fastq"))
-        if store and self.hk.tag("samplesheet") is None:
-            self.hk.add_commit(self.hk.new_tag("samplesheet"))
-        if store and self.hk.tag(flow_cell_id) is None:
-            self.hk.add_commit(self.hk.new_tag(flow_cell_id))
-        stats_data: StatsFlowcell = self.stats.flowcell(flow_cell_id)
-        flow_cell: models.Flowcell = self.db.flowcell(flow_cell_id)
-
-        if flow_cell is None:
-            flow_cell: models.Flowcell = self.db.add_flowcell(
-                name=flow_cell_id,
-                sequencer=stats_data.sequencer,
-                sequencer_type=stats_data.sequencer_type,
-                date=stats_data.date,
-            )
-        flow_cell.status = "ondisk"
-
-        sample_sheet_path = self._sample_sheet_path(flow_cell_id)
-        if not Path(sample_sheet_path).exists():
-            LOG.warning(f"unable to find samplesheet: {sample_sheet_path}")
-        elif store:
-            self.store_samplesheet(flow_cell_id, sample_sheet_path)
+        self._add_tag_to_housekeeper(
+            store=store,
+            tags=[
+                SequencingFileTag.FASTQ,
+                SequencingFileTag.SAMPLE_SHEET,
+                flow_cell_id,
+                SequencingFileTag.CGSTATS_LOG,
+            ],
+        )
+        cgstats_flow_cell: StatsFlowcell = self.stats.flowcell(flowcell_name=flow_cell_id)
+        flow_cell: Flowcell = self._add_flow_cell_to_status_db(
+            cgstats_flow_cell=cgstats_flow_cell,
+            flow_cell=self.db.get_flow_cell(flow_cell_id=flow_cell_id),
+            flow_cell_id=flow_cell_id,
+        )
+
+        self._include_sample_sheet_to_housekeeper(
+            flow_cell_dir=flow_cell_dir, flow_cell_id=flow_cell_id, store=store
+        )
+        self._include_cgstats_log_to_housekeeper(
+            flow_cell_dir=flow_cell_dir, flow_cell_id=flow_cell_id, store=store
+        )
+        self._parse_flow_cell_samples(
+            cgstats_flow_cell=cgstats_flow_cell,
+            flow_cell=flow_cell,
+            flow_cell_id=flow_cell_id,
+            store=store,
+        )
+        return flow_cell
 
-        for sample_data in stats_data.samples:
-            LOG.debug(f"adding reads/fastqs to sample: {sample_data.name}")
-            sample_obj = self.db.sample(sample_data.name)
-            if sample_obj is None:
-                LOG.warning(f"unable to find sample: {sample_data.name}")
+    def _add_tag_to_housekeeper(self, store: bool, tags: List[str]) -> None:
+        """Add and commit tag to Housekeeper if not already exists in database."""
+        for tag in tags:
+            if store and self.hk.tag(name=tag) is None:
+                self.hk.add_commit(self.hk.new_tag(tag))
```

Comment:
It has the store parameter (from previous) so that anchors it here. But otherwise it could be defined in hk. This PR is already changing a lot. I would like to not go out of scope more than needed now.